test/test_fcntl.rs

```diff
@@ -49,15 +49,49 @@ fn test_readlink() {
 mod linux_android {
     use std::io::prelude::*;
     use std::os::unix::prelude::*;
+    use std::io::SeekFrom;
 
     use libc::loff_t;
 
-    use nix::fcntl::{SpliceFFlags, FallocateFlags, fallocate, splice, tee, vmsplice};
+    use nix::fcntl::{SpliceFFlags, FallocateFlags, fallocate, splice, tee, vmsplice, copy_file_range};
     use nix::sys::uio::IoVec;
     use nix::unistd::{close, pipe, read, write};
 
     use tempfile::{tempfile, NamedTempFile};
 
+    /// This test creates a temporary file containing the contents
+    /// 'foobarbaz' and uses the `copy_file_range` call to transfer
+    /// 3 bytes at offset 3 (`bar`) to another empty file at offset 0. The
+    /// resulting file is read and should contain the contents `bar`.
+    /// The from_offset should be updated by the call to reflect
+    /// the 3 bytes read (6).
+    ///
+    /// Fix me: test is disabled for linux based builds, because Jenkins
+    /// Linux version is too old for `copy_file_range`. Should work
+    /// on netbsd build.
+    #[test]
+    #[cfg_attr(target_env = "linux", ignore)]
```

Comment:
Linux is a target_os, not a target_env.  Also, you should leave a message with the ignore attribute.  And it's not going to work on a "netbsd build", because NetBSD doesn't have this syscall at all.  Finally, we use Travis, not Jenkins.

Reply:
Is there anything else that is holding this PR back?

Reply:
Yes.  The OP needs to address my last comments, fix the CI failures, and rebase.